### DISCUSSION TITLE
[@lit-labs/react] React-wrapped lit component doesn't render

### DISCUSSION THREAD

**Discussion by kws1207** [lit/lit#3552] [Q&A] [answered]
Hello, I'm struggling using `@lit-labs/react`. 

In my [minimal reproducible example](https://github.com/kws1207/lit-react-test), both storybook and react webpage doesn't properly render my `CounterButton` component, instead below error appears.

> The following properties on element counter-button will not trigger updates as expected because they are set using class fields: title, counter. Native class fields and some compiled output will overwrite accessors used for detecting changes. See https://lit.dev/msg/class-field-shadowing for more information.

Below is the code of my `CounterButton` component.

```ts
import { createComponent } from "@lit-labs/react";
import { html, css, LitElement } from "lit";
import { customElement, property } from "lit/decorators.js";
import React from "react";

@customElement("counter-button")
class CounterButtonWC extends LitElement {
  static styles = css`
    :host {
      display: block;
      padding: 25px;
      color: var(--counter-button-text-color, #000);
    }
  `;

  @property({ type: String }) title = "Hey there";

  @property({ type: Number }) counter = 5;

  __increment() {
    this.counter += 1;
  }

  render() {
    return html`
      <h2>${this.title} Nr. ${this.counter}!</h2>
      <button @click=${this.__increment}>increment</button>
    `;
  }
}

export const CounterButton = createComponent({
  react: React,
  tagName: "counter-button",
  elementClass: CounterButtonWC,
});

```

Am I doing something wrong? Any kind of help would be appreciated.

Thank you.

**Comment by augustjk** [comment, marked as answer]
Hello! The problem seems to be that Create React App does not support decorators. https://create-react-app.dev/docs/can-i-use-decorators

Instead of decorators, using `static properties = ` and `customElements.define()` fixes it.

If you'd like to use decorators, you'd need to either eject from CRA or find an alternate way to adjust its build configuration, or use a different tool for bootstrapping the react project like vitejs.dev, or manually set up the project.
